engine/worker/requirement.go

```diff
@@ -158,24 +158,32 @@ func checkServiceRequirement(w *currentWorker, r sdk.Requirement) (bool, error)
 }
 
 func checkMemoryRequirement(w *currentWorker, r sdk.Requirement) (bool, error) {
-	// If we are in a docker and in a provisioned worker don't run job in this worker
-	if w.model.Type == sdk.Docker && w.bookedWJobID == 0 {
-		return false, nil
-	}
-
-	v, err := mem.VirtualMemory()
+	var totalMemory int64
+	neededMemory, err := strconv.ParseInt(r.Value, 10, 64)
 	if err != nil {
 		return false, err
 	}
-	totalMemory := v.Total
 
-	neededMemory, err := strconv.ParseInt(r.Value, 10, 64)
-	if err != nil {
-		return false, err
+	switch w.model.Type {
+	// Check env variables in a docker is safer than mem.VirtualMemory
+	case sdk.Docker:
+		var err error
+		memoryEnv := os.Getenv("CDS__MEMORY")
```

Comment:
CDS__MEMORY contains the memory = spawnArgs.Model.ModelDocker.Memory, so the memory required by the worker model.
here, we compare this memory required by the worker model with the memory of the requirement set by user.
Can you add comment about that in this func? I had to read the set of CDS__MEMORY to understand this check.